api_app/views.py

```diff
@@ -107,7 +94,81 @@ def _analysis_request(
     )
     ser.is_valid(raise_exception=True)
 
-    response_dict = ser.data
+    response_dict = {
+        "count": 1,
+        "results": [ser.data],
+    }
+
+    logger.debug(response_dict)
+
+    return Response(
+        response_dict,
+        status=status.HTTP_200_OK,
+    )
+
+
+def _multi_observable_analysis_request(
+    request,
+    serializer_class: ObservableAnalysisSerializer,
+):
+    """
+    Prepare and send multiple observables for analysis
+    """
+    logger.info(
+        f"_analysis_request {serializer_class} received request from {request.user}."
+        f"Data:{dict(request.data)}."
+    )
+
+    # serialize request data and validate
+    serializer = serializer_class(data=request.data, many=True)
+    serializer.is_valid(raise_exception=True)
+
+    serialized_data = serializer.validated_data
+    print(serialized_data)
+    runtime_configurations = [
+        data.pop("runtime_configuration", {}) for data in serialized_data
+    ]
+
+    # save the arrived data plus new params into a new job object
+    jobs = serializer.save(
+        user=request.user,
+    )
+
+    logger.info(f"New Jobs added to queue <- {repr(jobs)}.")
+
+    # Check if task is test or not
+    if not settings.STAGE_CI:
+        # fire celery task
+        for index, job in enumerate(jobs):
+            runtime_configuration = runtime_configurations[index]
+            celery_app.send_task(
+                "start_analyzers",
+                kwargs=dict(
+                    job_id=job.pk,
+                    analyzers_to_execute=job.analyzers_to_execute,
+                    runtime_configuration=runtime_configuration,
+                ),
+            )
+
+    ser = AnalysisResponseSerializer(
+        data=[
+            {
+                "status": "accepted",
+                "job_id": job.pk,
+                "warnings": serialized_data[index]["warnings"],
+                "analyzers_running": job.analyzers_to_execute,
+                "connectors_running": job.connectors_to_execute,
+            }
+            for index, job in enumerate(jobs)
+        ],
+        many=True,
+    )
+    ser.is_valid(raise_exception=True)
+
+    response_dict = {
+        "count": len(ser.data),
+        "results": ser.data,
+    }
```

Comment:
I have to admit that I do not like to have 2 endpoints that share 95% of their code. The only difference that this endpoint has with `_analysis_request` is the `many=True` and the iteration on the `jobs`.
I think i would prefer a single method that manages both cases and, if necessary (meaning that you can't discern if you have `many=True` or not just by checking `request.data`) another argument of the function that is used to distinguish between the two cases

Reply:
One endpoint is left as it is for backward compatibility to be removed in the future version

Reply:
For what reason we should remove the other endpoint? If we can avoid breaking changes at any time, is better.
Other than that, I talked about `endpoint`, but what i mean here are the functions `_multi_observable_analysis_request` and `_analysis_request`.

Reply:
I agree with Simone. With a little more effort and following the implementation he suggested, we could have a more serious result.

Reply:
Is the intended result having a single endpoint handle both cases? In that case, one endpoint would have two very different possible response structures. This seems to violate the REST API conventions as all documentation tools expect endpoints' responses to have a fixed structure

Reply:
```python3
def _observable_analysis_request(request,
    serializer_class: ObservableAnalysisSerializer,
    many:bool=True
    # do stuff
    return something
    
@add_docs(
    description="This endpoint allows to start a Job related to an observable. "
    "Retained for compatibility",
    request=ObservableAnalysisSerializer,
    responses={200: AnalysisResponseSerializer},
)
@api_view(["POST"])
def analyze_observable(request):
    res =  _observable_analysis_request(request, ObservableAnalysisSerializer)
    # do parse
    return res


@add_docs(
    description="This endpoint allows to start Jobs related to multiple observables",
    request=ObservableAnalysisSerializer,
    responses={200: AnalysisResponseSerializer},
)
@api_view(["POST"])
def analyze_multiple_observables(request):
    res =  _observable_analysis_request(request, ObservableAnalysisSerializer)    
      # do parse
    return res
    
 ```
 This is the structure of what i expected

Reply:
@0ssigeno Pushed. Does it look good?

Reply:
I like it!